### PR TITLE
New Interactive Graph Axis Tick Labels (rendered with the MathJax font)

### PR DESCRIPTION
**Pull request by SonicScrewdriver**
## Summary:
This PR adds a new approach to rendering our interactive graph axis tick labels. In order to ensure that we can support cases where the [tickSteps are less than the gridSteps](https://khanacademy.atlassian.net/browse/LEMS-1890), we're now manually rendering these labels. 

The labels have also been updated to render using the `MathJax_Main-Regular` (or `MJXTEX`) font. 

NOTE: There's still a bug regarding overlapping labels on the initial negative axis points, but that will be tackled in [LEMS-1891](https://khanacademy.atlassian.net/browse/LEMS-1891). 

Issue: LEMS-1878

## Test plan:
- manual testing in flipbook using all edge cases 
- manual testing in webapp to ensure the fonts are still correctly identified 

## Screen shots:
Screenshot 2024-04-10 at 4 18 44 PM

Screenshot 2024-04-10 at 4 19 34 PM

Screenshot 2024-04-10 at 4 20 05 PM

[LEMS-1891]: https://khanacademy.atlassian.net/browse/LEMS-1891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/perseus/src/widgets/interactive-graphs/grid.tsx [hunk 1]
```diff
@@ -8,12 +8,18 @@ import type {GraphRange} from "../../perseus-types";
 import type {SizeClass} from "../../util/sizing-utils";
 import type {vec} from "mafs";
 
+/*
```
Comment (SonicScrewdriver) [inline packages/perseus/src/widgets/interactive-graphs/grid.tsx:11]: I added this comment for my own sanity, and wondered if I should keep it. Happy to remove it however!

File: packages/perseus/src/widgets/interactive-graphs/axis-ticks.tsx [hunk 2]
```diff
@@ -24,6 +24,21 @@ const YGridTick = ({y}: {y: number}) => {
...
             />
+            {
+                // TODO (LEMS-1891): Negative one is a special case as the labels can
+                // overlap with the axis line. We should handle this case more gracefully.
+            }
+            {y !== -1 && (
+                <text
+                    height={20}
+                    width={50}
+                    textAnchor="end"
```
Comment (SonicScrewdriver) [inline packages/perseus/src/widgets/interactive-graphs/axis-ticks.tsx:35]: We want "end" here as we want to right-align the y axis labels.

File: packages/perseus/src/widgets/interactive-graphs/axis-ticks.tsx [hunk 3]
```diff
@@ -41,6 +56,21 @@ const XGridTick = ({x}: {x: number}) => {
...
             />
+            {
+                // TODO (LEMS-1891): Negative one is a special case as the labels can
+                // overlap with the axis line. We should handle this case more gracefully.
+            }
+            {x !== -1 && (
+                <text
+                    height={20}
+                    width={50}
+                    textAnchor="middle"
```
Comment (SonicScrewdriver) [inline packages/perseus/src/widgets/interactive-graphs/axis-ticks.tsx:67]: We want middle here, as we want to ensure the x axis tick labels are centered under their appropriate 'tick'.

File: packages/perseus/src/widgets/interactive-graphs/grid.tsx [hunk 4]
```diff
@@ -8,12 +8,18 @@ import type {GraphRange} from "../../perseus-types";
...
+gridStep: Where the lines on the grid will render.
+tickStep: Where the little black axis lines & labels (ticks) should render. (e.g. 1, 2, 3, 4, 5)
+snapStep: Where the graph points will lock to when they are dragged.
+*/
```
Comment (SonicScrewdriver) [inline packages/perseus/src/widgets/interactive-graphs/grid.tsx:15]: I added these comments for my own sanity, and wondered if I should keep them. Happy to delete this if it's not useful however.

Comment (jeremywiebe) [reply packages/perseus/src/widgets/interactive-graphs/grid.tsx:15]: I love that these exist, please don't remove them entirely. 

But, could these go on the `GridProps` type itself?  Or some other type where these keys are defined that we could document?

File: packages/perseus/src/widgets/interactive-graphs/mafs-styles.css [hunk 5]
```diff
@@ -124,3 +124,8 @@
 .MafsView .movable-point:focus-visible .movable-point-focus-outline {
     visibility: visible;
 }
+
+.MafsView .axis-ticks text {
+    font-size: 14px;
+    font-family: "MJXTEX";
```
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/mafs-styles.css:130]: Where is this font family defined? Will it always be available?

I kind of thought we'd have to add a font-face definition like:

```css
@font-face {
  font-family: MJXTEX;
  src: url("https://cdn.kastatic.org/fonts/mathjax/MathJax_Main-Regular.woff") format("woff");
}
```

But maybe that definition already exists elsewhere in Perseus?

If we don't have an explicit font-face definition, then I think the correct font will only be available if MathJax has rendered an expression since the last full-page refresh.

**Review by benchristel — COMMENTED** [review]
It looks SO GOOD!

I left one question inline about how the font face gets loaded.

File: packages/perseus/src/widgets/interactive-graphs/mafs-styles.css (hunk 5 above)
Comment (benchristel) [reply packages/perseus/src/widgets/interactive-graphs/mafs-styles.css:130]: If we end up defining the font-face in Perseus CSS, we should probably call it something other than `MJXTEX` to avoid any possible conflict with the font-face that MathJax defines. They _should_ always be the same font, but you never know...

Comment (SonicScrewdriver) [reply packages/perseus/src/widgets/interactive-graphs/mafs-styles.css:130]: We found that it's actually already well defined in MathJax itself, which I believe gets pulled into index.css when we compile the dist. 

Screenshot 2024-04-11 at 2 51 24 PM

I made sure to test in webapp, and it loads in properly even if I go directly to the segment graph test page without loading any ancillary MathJax. :)

Comment (SonicScrewdriver) [reply packages/perseus/src/widgets/interactive-graphs/mafs-styles.css:130]: I believe @jeremywiebe thought this would be bundled properly after we did some digging! 

It also seems to load properly in flipbook, even when I make sure that there's no other Mathjax on the page at all.
 
Screenshot 2024-04-11 at 3 04 31 PM

I was unable to find any examples of content in webapp that don't _also_ end up using ancillary MathJax as they usually have to indicate points or some kind of value to set up the question. Even the segment testing page has some minor bits of MathJax floating around.🤔

Comment (benchristel) [reply packages/perseus/src/widgets/interactive-graphs/mafs-styles.css:130]: It does indeed seem like MathJax sneaks in in a lot of places!

However, I'm still wary of relying on MathJax being the math renderer that is used in production. We allow clients of Perseus to inject any TeX component, and I suspect that third-party users of Perseus (e.g. [Learning Equity](https://khanacademy.slack.com/archives/CJDRXTGQ7/p1712852640716959)) are still using KaTeX.

I tried out switching to a custom `@font-face` and it works. The fonts are only loaded once if we use the same CDN URL that the MathJax renderer is configured to use:

```css
@font-face {
    font-family: Mafs-MJXTEX;
    src: url("https://cdn.kastatic.org/fonts/mathjax/MathJax_Main-Regular.woff") format("woff");
}

.MafsView .axis-ticks text {
    font-size: 14px;
    font-family: "Mafs-MJXTEX";
}
```

Comment (SonicScrewdriver) [reply packages/perseus/src/widgets/interactive-graphs/mafs-styles.css]: Lovely, that seems safer! I've implemented your suggestion, thank you greatly. :)

File: packages/perseus/src/widgets/interactive-graphs/axis-ticks.tsx [hunk 6]
```diff
@@ -63,10 +97,12 @@ export function generateTickLocations(
 type Props = {
     tickStep: [number, number];
     range: [[number, number], [number, number]];
+    graphSize: vec.Vector2;
```
Comment (benchristel) [inline packages/perseus/src/widgets/interactive-graphs/axis-ticks.tsx:100]: `graphSize` doesn't seem to be used. I think we can remove it?

**Review by benchristel — APPROVED** [review]
LGTM! Just one small thing—I think the `graphSize` prop is unused and can be removed.
